### ISSUE TITLE
How to optimize training and avoid overfitting for a new dataset

### ISSUE THREAD

**Issue by imdusa** [open-mmlab/mmskeleton#274]
Thanks for the great share, I am using your repository to work with another dataset where the skeletons are not humans ( I read the dataset in by customizing the feeder etc. and I also changed the joints etc.) It is a smaller dataset, and when I train the training loss keeps going down however the validation accuracy is very low, it is around 57-62 at best during early epochs, and then it gets worse and settles around 57 at best or lower around 40. 

Firstly, for this dataset (there are two actors and each has 5 joints), how can I optimize the parameters? Which ones shall I experiment with? I have clipped videos into segments of 90 frames which equals to 3 seconds. 

And secondly, what measures can I take to avoid overfitting?

I believe your method intuitively makes a lot of sense so I really want this to work out, I would appreciate any thoughts, insights.

**Comment by luoluo-gif** [comment]
I met the same question.If you solve them,can you tell me the answer of these. I would be appreciated it~

**Comment by imduygusarikaya** [comment]
@luoluo-gif 
I am still trying to make the network perform better and decrease the validation loss further, but I was not able to. I managed to get 67% accuracy (due to a mistake I made with data reading which I fixed now)

But here are things I tried but did not improve the accuracy:
using adam optimizer
using adam optimizer with a scheduler that chnages learning rate when there is a plateu
data augmentation ( random flip, crop, shear, scale, rotation) -> Somehow data augmentation made things even worse, I also tried original data+augmented data and it performed much worse
I tried a bunch of variables for learning rates and steps but no luck

**Comment by a726** [comment]
> Thanks for the great share, I am using your repository to work with another dataset where the skeletons are not humans ( I read the dataset in by customizing the feeder etc. and I also changed the joints etc.) It is a smaller dataset, and when I train the training loss keeps going down however the validation accuracy is very low, it is around 57-62 at best during early epochs, and then it gets worse and settles around 57 at best or lower around 40.
> Firstly, for this dataset (there are two actors and each has 5 joints), how can I optimize the parameters? Which ones shall I experiment with? I have clipped videos into segments of 90 frames which equals to 3 seconds.
> And secondly, what measures can I take to avoid overfitting?
> I believe your method intuitively makes a lot of sense so I really want this to work out, I would appreciate any thoughts, insights.

Hello, may I ask how do you use openpose saving bone tag other data set?

**Comment by d0683228** [comment]
Hi @imduygusarikaya 
Did you solve the problem?
I met the same problem and I tried chage some hyperparameter
change learning rate ,batchsize ,set dropout
but still don't work

**Comment by nanhui69** [comment]
> @luoluo-gif
> I am still trying to make the network perform better and decrease the validation loss further, but I was not able to. I managed to get 67% accuracy (due to a mistake I made with data reading which I fixed now)
> 
> But here are things I tried but did not improve the accuracy:
> using adam optimizer
> using adam optimizer with a scheduler that chnages learning rate when there is a plateu
> data augmentation ( random flip, crop, shear, scale, rotation) -> Somehow data augmentation made things even worse, I also tried original data+augmented data and it performed much worse
> I tried a bunch of variables for learning rates and steps but no luck

what process do you apply in make dataset ? could you share some code about preprocdess dataset?

**Comment by lakpa-tamang9** [comment]
Any updates here? Regarding decreasing the validation loss.
